R/generate_simulated_data.R

```diff
@@ -142,34 +139,36 @@ generate_simulated_data <- function(r_in_weeks = # nolint
   )
   model$expose_functions(global = TRUE)
 
-  # Pull parameter values into memory
+  # Get other variables needed for forward simulation ------------------------
   params <- get_params(input_params_path) # load in a single row tibble
-  par_names <- colnames(params) # pull them into memory
-  for (i in seq_along(par_names)) {
-    assign(par_names[i], as.double(params[i]))
-  }
+
+  # Get pop fractions of each subpop. There will n_sites + 1 subpops
+  pop_fraction <- c(
+    ww_pop_sites / pop_size,
+    (pop_size - sum(ww_pop_sites)) / pop_size
+  )
+  n_subpops <- length(pop_fraction)
 
   # Create a tibble that maps sites, labs, and population sizes of the sites
   n_sites <- length(unique(site))
-  site_lab_map <- data.frame(
+  site_lab_map <- create_site_lab_map(
```

Comment:
I would argue it is a good practice to be verbose about the passed arguments. In other words, unless the function has a single argument, I would explicitly 
`arg1 = my_arg1`.

Reply:
Ooh this is a good point, especially since there are a lot of these and order is constantly changing. Will go through and adjust so all of them are explicit